JSDR 86 (#130)
Macrometacorp/jsC8: GETTING_STARTED.md

  GDN and is not required as input unless you want to validate a document
  against its current revision.

Here is an example of a valid document:

```
    {
        '_id': 'students/bruce',
        '_key': 'bruce',
        '_rev': '_Wm3dzEi--_',
        'first_name': 'Bruce',
        'last_name': 'Wayne',
        'address': {
            'street' : '1007 Mountain Dr.',
            'city': 'Gotham',
            'state': 'NJ'
        },
        'is_rich': True,
        'friends': ['robin', 'gordon']
    }
```

**Edge documents (edges)** are similar to standard documents but with two
additional required fields `_from` and `_to`. Values of these fields must be the
handles of "from" and "to" vertex documents linked by the edge document in
question (see [Graph](docs/Reference/Graph/README.md) for details).

Here is an example of a valid edge document:

```
    {
        '_id': 'friends/001',
        '_key': '001',
        '_rev': '_Wm3dyle--_',
        '_from': 'students/john',
        '_to': 'students/jane',
        'closeness': 9.5
    }
```

---

## 🔐 Auth - Connect to GDN

The first step to start using GDN is establishing a connection to

> [play.macrometa.io](https://play.macrometa.io/). When this code executes, it

Reviewer: ```suggestion
[play.paas.macrometa.io](https://play.paas.macrometa.io/). When this code executes, it
```

Reviewer: Both play.paas and play work but since we have mentioned play.paas in website docs we can stick with that maybe

initializes the server connection to your nearest region. You can access your
Macrometa GDN account using several methods, such as
[API keys](https://macrometa.com/docs/account-management/api-keys/),
[User authentication](https://macrometa.com/docs/account-management/auth/user-auth),
[Token-based authentication (JWT)](https://macrometa.com/docs/account-management/auth/jwts).

### How to create and instance of jsc8?

```js
const jsc8 = require("jsc8");

> const client = new jsc8("https://gdn1.macrometa.io");

Reviewer: ```suggestion
const client = new jsc8("https://play.macrometa.io");
```

```

or to have fail-over support:

```js
const jsc8 = require("jsc8");
const client = new jsc8([

>   "https://gdn1.macrometa.io",

Reviewer: ```suggestion
  "https://play1.paas.macrometa.io",
```

>   "https://gdn2.macrometa.io",

Reviewer: ```suggestion
  "https://play2.paas.macrometa.io",
```

]);
```

This connection string actually represents the default value(

> `"https://gdn1.macrometa.io"` ), so if this is the value you want, you can omit

Reviewer: If this is the default then we need to change in code and here too to maybe play.paas.macrometa.io

> `"https://gdn1.macrometa.io"` ), so if this is the value you want, you can omit

Reviewer: ```suggestion
`"https://play.paas.macrometa.io"` ), so if this is the value you want, you can omit
```

> `"https://play.paas.macrometa.io"` ), so if this is the value you want, you can omit

Reviewer: @dlozina-macrometa we need to check the default value and change it to play.paas if not already present

const client = new jsc8();
```

If that’s still too verbose for you, you can invoke the driver directly using
the require statement itself.

```js
const client = require("jsc8")();
```

The outcome of the three calls should be identical.

### Auth with email and password

To start working, you first have to log in. This gets the auth token and is
added in each API call. Continuing from the previous snippet, the following
statement will log you into the c8 using the provided credentials.

```js
async function login() {
  return await client.login(email, password);
}
```

Below line should print the jwt token. However, this is just for demonstrative
purposes as this token will be added to further requests. Simple client.login
should suffice for most cases.

```js
login().then(console.log);
```

To use the demotenant, we need to use client.useTenant as below:

```js
client.useTenant(tenant_name);
```

### Auth with token

If you want to log in with token then you just need to pass the token wrapping
in Object with your region URL with specific fabric.

**Note**

> Default value for fabric is `_system`

```js
const jsc8 = require("jsc8");
client = new jsc8({

>   url: "https://gdn1.macrometa.io",

Reviewer: ```suggestion
  url: "https://play.paas.macrometa.io",
```

  token: "XXXX",
  fabricName: "_system",
});
```

or to have fail-over support:

```js
const jsc8 = require("jsc8");
client = new jsc8({

>   url: ["https://gdn1.macrometa.io", "https://gdn2.macrometa.io"],

Reviewer 2: GDN should be changed to play federation.

Reviewer: Yes Yes I will change it everywhere
